lib/chef/shell.rb

```diff
@@ -295,18 +296,19 @@ def parse_opts
     private
 
     def config_file_for_shell_mode(environment)
+      dot_chef_dir = Chef::Util::PathHelper.home('.chef')
       if config[:config_file]
         config[:config_file]
-      elsif environment && ENV['HOME']
+      elsif environment && dot_chef_dir
```

Comment:
ENV['HOME'] can be artificially set when the directory doesn't exist but I believe I went through all the instanced modified to check if it makes more sense if we should use an environment variable if that directory doesn't exist.  In this one instance, I did wonder whether the best check would actually be "elsif environment" and not check dot_chef_dir at all but again...  unsure.